extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -44,10 +48,31 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
 
     /** @param {!ShoppingConfigDataDef} tagData */
     this.tagData_ = null;
+
+    /** @param {boolean} element */
+    this.hasShoppingAttachment_ = false;
+
+    /** @param {!AmpElement} element */
+    this.shoppingTagEl_ = null;
   }
 
   /** @override */
   buildCallback() {
+    const pageElement = closestAncestorElementBySelector(
```

Comment:
Nit: please add a comment that states that this prevents the component from building if there is no shopping attachment.